robot/rospackages/src/mcu_control/scripts/MockPdsNode.py

```diff
@@ -0,0 +1,276 @@
+
+        #Get 1 noise immediately while ros is running
+        currentNoise = get_states_values("PDS_mock_global_noise", noiseValues, currentNoise)
+
+        #Get 1 voltage incrementally while ros is running
+        if (currentVoltage is not None):
+            currentVoltage = get_parameter_value("PDS_mock_voltage", voltages, currentVoltage, currentNoise, voltageRate/rospyRate)
+
+        #Get 3 temperatures incrementally while ros is running
+        for x in range(len(currentTemps)):
+            if (currentTemps[x] is not None):
+                currentTemps[x] = get_parameter_value(parameterTemps[x], temps, currentTemps[x], currentNoise, tempRate/rospyRate)
+
+        #Get 6 wheel currents incrementally while ros is running
+        for x in range(len(currentWheelCurrents)):
+            if (currentWheelCurrents[x] is not None):
+                currentWheelCurrents[x] = get_parameter_value(parameterCurrents[x], currents, currentWheelCurrents[x], currentNoise, currentRate/rospyRate)
+
+        #Set values to objects which will get published
+        #Set voltage 1
+        if (currentVoltage is not None):
+            voltage.data = float(currentVoltage)
+        #Set temperatures 1-3
+        if (currentTemps[0] is not None):
+            thermistorTemps.therm1 = float(currentTemps[0])
+        if (currentTemps[1] is not None):
+            thermistorTemps.therm2 = float(currentTemps[1])
+        if (currentTemps[2] is not None):
+            thermistorTemps.therm3 = float(currentTemps[2])
+
+        wheelCurrents.effort.clear()
+        #Set wheel currents 1-6
+        for x in range(len(currentWheelCurrents)):
+            if(currentWheelCurrents[x] is not None):
+                wheelCurrents.effort.append(float(currentWheelCurrents[x]))
+
+        #Publish values to topics (Voltage, ThermistorTemps and Currents)
+        pubVoltage.publish(voltage)
+        pubTemp.publish(thermistorTemps)
+        pubWheelCurrent.publish(wheelCurrents)
+
+        #Add delay before each iteration of loop which runs rospy
+        rate.sleep()
+
+if __name__ == '__main__':
+    """Calls publisher function and passes parameter settings
+    Change the values for the parameter states (rise, stable and fall)
+    However, change the initial starting values for the parameters directly in publish_mock_data
+    No other places require to be modified to change the state (rise, stable, and fall) values"""
+
+    #Set voltages rise, stable and fall to be parametrized
+    voltages = {"rise": 20, "stable": 15, "fall": 10}
+
+    #Set temperatures rise, stable and fall to be parametrized
+    temps = {"rise": 100, "stable": 25, "fall": -20}
+
+    #Set temperatures rise, stable and fall to be parametrized
+    currents = {"rise": 1, "stable": 0.3, "fall": 0}
+
+    #Set temperatures rise, stable and fall to be parametrized
+    noiseValues = {"rise": 0.1, "stable": 0.05, "fall": 0.01}
```

Comment:
this doesnt make sense as we discussed. The fact that the noise is small isnt falling. The noise value should be a float not a string in the parameters. Also can we have noise for voltage, noise for currents and noise for temps. Because a noise of 0.05 is huge for currents lol but its appropriate for temps and voltage

Reply:
Ah yes, i completely forgot about that. So i take noiseVoltage, noiseCurrent, noiseTemps, and all of which are set using a float as opposed to a state (rise, stable, fall).

Reply:
very cool

Reply:
fixed